core/app/models/spree/adjustment.rb

```diff
@@ -72,9 +52,55 @@ class Adjustment < Spree::Base
     extend DisplayMoney
     money_methods :amount
 
+    def finalize!
+      update_attributes!(finalized: true)
+    end
+
+    def unfinalize!
+      update_attributes!(finalized: false)
+    end
+
+
+    # Deprecated methods
+    def state
+      ActiveSupport::Deprecation.warn "Adjustment#state is deprecated. Instead use Adjustment#finalized?", caller
+      finalized?? "closed" : "open"
+    end
+
+    def state=(new_state)
+      ActiveSupport::Deprecation.warn "Adjustment#state= is deprecated. Instead use Adjustment#finalized=", caller
+      case new_state
+      when "open"
+        self.finalized = false
+      when "closed"
+        self.finalized = true
+      else
+        raise "invaliid adjustment state #{new_state}"
+      end
+    end
+
+    def open?
+      ActiveSupport::Deprecation.warn "Adjustment#open? is deprecated. Instead use Adjustment#finalized?", caller
+      !closed?
+    end
+
     def closed?
-      state == "closed"
+      ActiveSupport::Deprecation.warn "Adjustment#closed? is deprecated. Instead use Adjustment#finalized?", caller
+      finalized?
+    end
+
+    def open
+      ActiveSupport::Deprecation.warn "Adjustment#open is deprecated. Instead use Adjustment#unfinalize!", caller
+      unfinalize!
```

Comment:
is it worth having `finalize`/`unfinalize` (non-bang versions) for `open`/`close`?